### DISCUSSION TITLE
TaskIQ in production

### DISCUSSION THREAD

**Discussion by tyleralbee** [taskiq-python/taskiq#277] [Q&A] [answered]
What does a good production setup look like with TaskIQ? 

I did see in [another thread](https://github.com/orgs/taskiq-python/discussions/260#discussioncomment-7817074) that @s3rius suggested using [Helm](https://github.com/helm/helm) to manage Kubernetes applications, saying this:

> Personally I use helm charts for my applications and both taskiq workers and schedulers are declared using the same deployment template, but with different command.

I realize there is no one-size-fits-all answer to this question, but I'd appreciate some advice from more experienced developers. I'm trying to figure out how to host the broker/workers (ideally co-located with Redis) with the ability to scale up/down workers depending on load. In my case, there's one task that can take 5+minutes. 

I have enjoyed working with TaskIQ and I appreciate the quality of work put in by the maintainers.

**Comment by devraj** [comment, marked as answer]
Seeing I am the guy who started that thread, I thought I'd chime in 😀 and add my two cents worth. I have got it all working via a `helm` chart as described on the thread, it's a normal deployment which you should be able to scale up and down with demand (note that my chart is spinning up a worker, not a scheduler - which shouldn't be difficult but be weary that you shouldn't have multiple of them).

The chart [is available here](https://github.com/anomaly/lab-python-server/tree/main/charts/worker) I am yet to document or finalise some loose ends.

If it is of assistance then I am willing to write out my finding in a document and publish it on our experimental repo.

**Reply by tyleralbee** [reply]
That would be awesome! Your link to the chart is already very helpful. Thanks a ton.
